src/System.IO/ref/System.IO.cs

```diff
@@ -164,6 +164,11 @@ public partial class StreamReader : System.IO.TextReader
         public StreamReader(System.IO.Stream stream, System.Text.Encoding encoding, bool detectEncodingFromByteOrderMarks) { }
         public StreamReader(System.IO.Stream stream, System.Text.Encoding encoding, bool detectEncodingFromByteOrderMarks, int bufferSize) { }
         public StreamReader(System.IO.Stream stream, System.Text.Encoding encoding, bool detectEncodingFromByteOrderMarks, int bufferSize, bool leaveOpen) { }
+        public StreamReader(string path) { }
+        public StreamReader(string path, bool detectEncodingFromByteOrderMarks) { }
+        public StreamReader(string path, System.Text.Encoding encoding) { }
+        public StreamReader(string path, System.Text.Encoding encoding, bool detectEncodingFromByteOrderMarks) { }
+        public StreamReader(string path, System.Text.Encoding encoding, bool detectEncodingFromByteOrderMarks, int bufferSize) { }
```

Comment:
> merge all of System.IO.FileSystem into System.IO

It does not sound that bad to me. The alternative would be to construct the FileStream using reflection (cache delegate to `File.Open`) that sounds about as bad to me as merging it. I would probably do the reflection for now because of it is very easy to do - it won't be the first use for this purpose. Track it in #11151 and look at what to merge later once we have more complete picture.